observability.go

```diff
@@ -0,0 +1,148 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+const (
+	OtelMetricExportInterval = 1 * time.Minute
+	OtelTraceExportInterval  = 1 * time.Minute
+	OtelLogExportInterval    = 10 * time.Second
```

Comment:
I'm curious where these values come from, are they something that we've got set up somewhere already, or are they the defaults from upstream? i.e. how'd we land on these values?

Reply:
The default values are:
- logging: 1 second
- metrics: 60 seconds
- traces: 5 seconds

This is net new / not configured in wasmcloud host atm. 
IMO: Would rather not hardcode these. Either leave the defaults or take values from HostData.

Reply:
That seems totally reasonable